cpp/community_detection_module/algorithm_online/community_detection.hpp

```diff
@@ -55,62 +57,188 @@ class LabelRankT {
+    for (const auto node_id : deleted_nodes_ids) {
+      label_Ps.erase(node_id);
+      sum_w.erase(node_id);
+      times_updated.erase(node_id);
+    }
+  }
 
-  ///@brief Retrieves the total weight of edges between two nodes.
-  /// With undirected graphs it considers all edges between the given node pair.
+  ///@brief Retrieves the total weight of relationships between two nodes.
+  /// With undirected graphs it considers all relationships between the given node pair.
   /// On a directed graph, it checks if the source and target nodes match.
   ///
   ///@param from_node_id -- source node’s Memgraph ID
   ///@param to_node_id -- target node’s Memgraph ID
   ///
-  ///@return -- total weight of edges between from_node_id and to_node_id
-  double GetTotalWeightBetween(std::uint64_t from_node_id,
-                               std::uint64_t to_node_id) const;
+  ///@return -- total weight of relationships between from_node_id and to_node_id
+  double GetTotalWeightBetween(const mgp::Graph &graph, std::uint64_t from_node_id, std::uint64_t to_node_id) const {
+    double total_weight = 0;
+
+    for (const auto &out_relationship : graph.GetNodeById(mgp::Id::FromUint(from_node_id)).OutRelationships()) {
+      if (out_relationship.To().Id().AsUint() == to_node_id) {
+        total_weight += GetWeight(graph, out_relationship);
+      }
+    }
+    if (!is_directed) {
+      for (const auto &in_relationship : graph.GetNodeById(mgp::Id::FromUint(from_node_id)).InRelationships()) {
+        if (in_relationship.From().Id().AsUint() == to_node_id) {
+          total_weight += GetWeight(graph, in_relationship);
+        }
+      }
+    }
+
+    return total_weight;
+  }
+
+  ///@brief Retrieves given relationship’s weight.
+  double GetWeight(const mgp::Graph &graph, const mgp::Relationship &relationship) const {
+    if (!is_weighted) return DEFAULT_WEIGHT;
 
-  ///@brief Retrieves given edge’s weight.
-  double GetWeight(std::uint64_t edge_id) const;
+    try {
+      const auto maybe_weight = relationship[weight_property];
+      return maybe_weight.IsNumeric() ? maybe_weight.ValueNumeric() : DEFAULT_WEIGHT;
+    } catch (const std::exception &e) {  // default if no such property exists
+      return DEFAULT_WEIGHT;
+    }
+  }
 
   ///@brief Resets each node’s number of updates.
-  void ResetTimesUpdated();
+  void ResetTimesUpdated() {
+    for (auto &[_, count] : times_updated) {
+      count = 0;
+    }
+  }
 
   /// Returns given node’s current community label.
   /// If no label was found (e.g. with the min_value set too high),
   /// the assigned label number is -1.
   ///
   /// @return -- given node’s current community label
-  std::int64_t NodeLabel(std::uint64_t node_id);
+  std::int64_t NodeLabel(std::uint64_t node_id) {
+    std::int64_t node_label = -1;
+    double max_P = 0;
+
+    for (const auto [label, P] : label_Ps[node_id]) {
+      if (P > max_P || (P == max_P && static_cast<std::int64_t>(label) < node_label)) {
```

Comment:
why do you need here static_cast

Reply:
`node_label` is a signed int because I use -1 to represent the edge case of nonexistent labels